### DISCUSSION TITLE
Some questions

### DISCUSSION THREAD

**Discussion by CA10C3SAR** [mark-wiemer/ahkpp#90] [Q&A] [answered]
**Q1-** Is possible to display more than one line when you rest mouse over a function name?
Example:

```
; Some text
; Some text
; Remove duplicates inside an array
RemoveDuplicates(object) {
}
```
![image](https://user-images.githubusercontent.com/6724006/117565238-8cd7ef00-b086-11eb-8a05-815325104512.png)

It only displays the first line above the function.

**Q2-** Is possible to disable "goto definition" from going from a file to another?

Example:
I have 3 files open in vs code that contains the same function, when I hit goto definition, sometimes it goes to another file instead of the current one that I'm focused on. 

I think this happens when it does not find the function in the current file but finds it in another file currently opened inside vscode.

**Q3-** Is possible to jump to an error line with a hotkey? Like on the AutoHotkeyScite that you press F4 and it goes to the error line.

**Q4-** Any news regarding the bugs in auto-format?

**Comment by mark-wiemer** [comment, marked as answer]
## Q1

> Q1- Is possible to display more than one line when you rest mouse over a function name?
> Example:
> 
> ; Some text
> ; Some text
> ; Remove duplicates inside an array
> RemoveDuplicates(object) {
> }
> (image)
> 
> It only displays the first line above the function.

Not yet, if you'd like this functionality, please open an issue for tracking. It's a limitation of the way method header comments are interpreted, should be easy to improve, but there are a lot of bugs on the backlog :/

## Q2

> Q2- Is possible to disable "goto definition" from going from a file to another?
> 
> Example:
> I have 3 files open in vs code that contains the same function, when I hit goto definition, sometimes it goes to another file instead of the current one that I'm focused on.
> 
> I think this happens when it does not find the function in the current file but finds it in another file currently opened inside vscode.

Also no. I don't follow your exact use case here: do you have the same file open in three editors, or do you have three separate files open that all have different functions with the same name? Do "go to definition" ever take you to the wrong file?

## Q3

> Q3- Is possible to jump to an error line with a hotkey? Like on the AutoHotkeyScite that you press F4 and it goes to the error line.

VS Code has built-in "Go to problem" commands that accomplish what you want. Unfortunately, this extension doesn't currently mark syntax errors, so those commands are not used for AHK :/ if you find an extension that does highlight errors, then "Go to next problem" (F8) should accomplish what you're asking.

## Q4

> Q4- Any news regarding the bugs in auto-format?

Many of them will require a complete overhaul to fix. Currently formatting is done in a very limited way: one line at a time with regular expressions. For more functionality, we should build an abstract syntax tree by parsing the entire file as we format. There are significant challenges with this approach, so it will take some time to get started.

## Conclusion

I'm sorry for the disappointing answers. I have been very busy recently and have not had time to work on this project. I have a few more tasks for the day, and then I hope to start working on a bug for the first time in a while. I do think there's a lot of potential here, and I want to keep chipping away at bugs as I make time for them. But there's a lot to figure out here (I'm very new to AHK and this is my first VS Code extension) and I'm just one person, so I appreciate your patience as I learn :)

If you want to track progress on any issue, please open one bug per issue and I'll add it to the backlog.

**Comment by kyklish** [comment]
> **Q3-** Is possible to jump to an error line with a hotkey? Like on the AutoHotkeyScite that you press F4 and it goes to the error line.

[AutoHotkey Debug](https://github.com/zero-plusplus/vscode-autohotkey-debug/wiki/Debug-console) extension with external library shows in debug console clickable link with launch-time or runtime error.

> **Q4-** Any news regarding the bugs in auto-format?

I had some Pull Request to fix several bugs. Waiting for approval.

**Reply by CA10C3SAR** [reply]
Does your forked branch contain these fixes?

**Reply by kyklish** [reply]
Yes, but they are in separate brunches. Some fixes are depends on other fixes.

**Reply by kyklish** [reply]
My main branch doesn't have fixes, each fix in separate brunch.
